metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -89,16 +85,27 @@ final class RenameProvider(
           .topMethodParents(occurence.symbol, semanticDb)
         txtParams <- if (parentSymbols.isEmpty) List(textParams)
         else parentSymbols.map(toTextParams)
-        currentReferences = referenceProvider.references(
-          toReferenceParams(txtParams)
-        )
+        isLocal = occurence.symbol.isLocal
+        currentReferences = referenceProvider
+          .references(
+            // we can't get definition by name for local symbols
+            toReferenceParams(txtParams, includeDeclaration = isLocal),
+            // local symbol will not contain a proper name
+            strictCheck = !isLocal
```

Comment:
Can you clarify what `strictCheck = true` does?

Reply:
I added a check that looks into what is underneath a range - anything there should be contained in the symbol itself. So string `Animal` should be contained in symbol `a/Animal#` 

Not the best solution, but I haven't figured out how else to filter out the odd reference under `@JsonCodec`

Reply:
Also I do the check only for renames - I think it's better to sacrifice a it of performance for correctness.

Reply:
Sounds like a good idea! How does this play with token edit distance? I thought we already kind of do something like this 🤔 Otherwise, how about renaming `strictCheck` into `checkMatchesText`?

Reply:
Sure, I will also check with the edit distances - might not work currently because of that

Reply:
I checked - token distance is not an issue since we use the reference range to find the name in the snapshot. Token distance is taken into account later on.

This is also tested in the RenameLspSuite since we simulate an edit there.

Reply:
Renamed to `checkMatchesText`